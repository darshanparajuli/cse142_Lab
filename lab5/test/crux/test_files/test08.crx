// test logical and

func andInt(a:int, b:int) : bool {
  return a and b;
}

func andFloat(a:float, b:float) : bool {
  return a and b;
}

func andBool(a:bool, b:bool) : bool {
  return a and b;
}

func main() : void {
  ::printBool(true);
}
