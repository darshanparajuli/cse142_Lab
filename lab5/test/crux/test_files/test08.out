Error type-checking file.
TypeError(4,12)[Cannot compute int and int.]
TypeError(4,3)[Function andInt returns bool not ErrorType(Cannot compute int and int.).]
TypeError(8,12)[Cannot compute float and float.]
TypeError(8,3)[Function andFloat returns bool not ErrorType(Cannot compute float and float.).]

